helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -443,8 +494,13 @@ private void calculateAndUpdateBaseline(ClusterModel clusterModel, RebalanceAlgo
     } else {
       LOG.debug("Assignment Metadata Store is null. Skip persisting the baseline assignment.");
     }
-    _globalBaselineCalcLatency.endMeasuringLatency();
+    _baselineCalcLatency.endMeasuringLatency();
     LOG.info("Finish calculating the new baseline.");
+
+    if (isbaselineUpdated && doSchedulePartialRebalance) {
+      LOG.info("Schedule a new rebalance after the new baseline calculated.");
+      RebalanceUtil.scheduleOnDemandPipeline(clusterName, 0l, false);
```

Comment:
Question about this: If we schedule an OnDemandPipeline, wouldn't this also trigger another global baseline calculation as well as a partial rebalance?

I am worried that this might actually become an infinite loop that triggers global baseline calculation non-stop... Or is that false flag preventing that?

Also, nit: `0l -> 0L`?

Reply:
No, unless there are other cluster changes happened which triggers a baseline calculation anyway.

Reply:
0L it is now : )